leaderboard/enriching/enricher.go

```diff
@@ -90,18 +106,89 @@ func (e *enricherImpl) Enrich(ctx context.Context, tenantID, leaderboardID strin
 		return nil, fmt.Errorf("could not unmarshal webhook response: %w", errors.Join(err, ErrEnrichmentCall))
 	}
 
-	return protoToMemberModels(result.Members), nil
+	metadataMap := make(map[string]map[string]string)
+	for _, m := range result.Members {
+		metadataMap[m.Id] = m.Metadata
+	}
+
+	for _, m := range members {
+		if data, ok := metadataMap[m.PublicID]; ok {
+			m.Metadata = data
+		}
+	}
+
+	return members, nil
+}
+
+func (e *enricherImpl) enrichWithCloudSave(ctx context.Context, tenantID string, members []*model.Member) ([]*model.Member, error) {
+	if e.config.CloudSave.Disabled[tenantID] {
+		return members, nil
+	}
+
+	ids := make([]string, len(members))
+	for i, m := range members {
+		ids[i] = m.PublicID
+	}
+
+	request := CloudSaveGetProfilesRequest{
+		TenantID: tenantID,
+		IDs:      ids,
+	}
+
+	jsonData, err := json.Marshal(request)
+	if err != nil {
+		return nil, fmt.Errorf("could not marshal request: %w", errors.Join(err, ErrEnrichmentInternal))
+	}
+
+	url, err := buildUrl(e.config.CloudSave.Url, cloudSaveEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("could not build cloud save url: %w", errors.Join(err, ErrEnrichmentInternal))
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewBuffer(jsonData))
+	if err != nil {
+		return nil, fmt.Errorf("could not create request: %w", errors.Join(err, ErrEnrichmentInternal))
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	raw, err := e.client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("could not complete request to cloud save: %w", errors.Join(err, ErrEnrichmentCall))
+	}
+	defer raw.Body.Close()
+
+	if raw.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("cloud save returned non OK response: %w", errors.Join(err, ErrEnrichmentCall))
+	}
+
+	res := &CloudSaveGetProfilesResponse{}
+	if err := json.NewDecoder(raw.Body).Decode(res); err != nil {
+		return nil, fmt.Errorf("could not unmarshal cloud save response: %w", errors.Join(err, ErrEnrichmentCall))
```

Comment:
Errors should be joined from the most close to the caller's layer to the most far, specific.

```suggestion
		return nil, fmt.Errorf("could not unmarshal cloud save response: %w", errors.Join(ErrEnrichmentCall, err))
```